### PR TITLE
[tock] rewrite GPIO to use autogenerated register map

### PR DESCRIPTION
**Pull request by semihalf-mazurek-michal**
### Pull Request Overview

This pull request replace GPIO register definitions
with auto-generated register map form OpenTitan project.

### Testing Strategy

Run TockOS kernel with Opentitan app hello.

### TODO or Help Wanted

N/A

### Documentation Updated

- [x] No updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Review by cfrantz — COMMENTED** [review]
This appears to be a single commit that has both the register updates and the driver modifications.  Can we separate this into two commits?

File: chips/lowrisc/src/gpio.rs [hunk 1]
```diff
@@ -1,122 +1,158 @@
...
+                    + MASKED_OE_UPPER::MASK.val(1u32 << upper_shift),
+            );
+        }
+    }
 
-        if self.gpio_registers.intr_state.is_set(pin) {
-            self.gpio_registers.intr_state.modify(pin.val(1));
+    pub fn handle_interrupt(&self) {
+        if self.gpio_registers.intr_state.is_set(self.pin) {
+            // Clear interupt (INTR_STATE has type rwi1c)
```
[the remark this answers is not in the document]
Comment (semihalf-mazurek-michal) [reply chips/lowrisc/src/gpio.rs:154]: https://docs.opentitan.org/hw/ip/gpio/doc/#Reg_intr_state
Documentations describe it as 'rw1c' maybe we have a typo in two place ?

File: chips/lowrisc/src/gpio.rs [hunk 2]
```diff
@@ -1,122 +1,158 @@
...
+            self.gpio_registers.intr_state.modify(self.pin.val(1));
```
[the remark this answers is not in the document]
Comment (semihalf-mazurek-michal) [reply chips/lowrisc/src/gpio.rs:155]: Replaced with write.

File: chips/lowrisc/src/gpio.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (semihalf-mazurek-michal) [reply chips/lowrisc/src/gpio.rs:154]: The same is stated in 'comparability' section for INTR_STATE.
https://docs.opentitan.org/doc/rm/comportability_specification/

File: chips/lowrisc/src/gpio.rs [hunk 3]
```diff
@@ -172,23 +157,23 @@ impl gpio::Configure for GpioPin<'_> {
         // TODO add checks for WRAL
         match mode {
             gpio::FloatingState::PullUp => {
-                self.pinmux_registers.dio_pad_attr[self.pin.shift]
+                self.pinmux_registers.dio_pad_attr[self.pin as usize]
```
[the remark this answers is not in the document]
Comment (semihalf-mazurek-michal) [reply chips/lowrisc/src/gpio.rs:160]: "I pretty certain that the way the pinmux is being programmed here is wrong (and was before you changed anything)."
I can't agree more ... As we discuss before i try to address this issue in next PR.

[the remark this answers is not in the document]
Comment (semihalf-mazurek-michal) [reply chips/lowrisc/src/gpio.rs:167]: I added a description of the problem with pinmux configuration to the code.

File: chips/lowrisc/src/gpio.rs [hunk 4]
```diff
@@ -1,122 +1,165 @@
...
+            Pin::Pin27 => INTR::GPIO_27,
+            Pin::Pin28 => INTR::GPIO_28,
+            Pin::Pin29 => INTR::GPIO_29,
+            Pin::Pin30 => INTR::GPIO_20,
+            Pin::Pin31 => INTR::GPIO_31,
+        }
+    }
+}
 
+// TODO Important fact about GPIO implementation quality.
```
Comment (semihalf-mazurek-michal) [inline chips/lowrisc/src/gpio.rs:60]: Added to reflect current code quality.

File: chips/lowrisc/src/gpio.rs [hunk 5]
```diff
@@ -126,59 +153,45 @@ impl<'a> GpioPin<'a> {
...
+        match self.is_set(&self.gpio_registers.direct_oe) {
             true => gpio::Configuration::InputOutput,
             false => gpio::Configuration::Input,
         }
     }
 
     fn set_floating_state(&self, mode: gpio::FloatingState) {
-        // There is unfortunately no documentation about how these
-        // registers map to the actual GPIOs, so just write all of them.
+        // TODO Add checks for all WRAL registers
```
[the remark this answers is not in the document]
Comment (semihalf-mazurek-michal) [reply chips/lowrisc/src/gpio.rs]: Fixed

**Comment by semihalf-mazurek-michal** [comment]
@cfrantz  LGTM

**Review by cfrantz — APPROVED** [review]

**Pull request merged**